website/docs/configuration/functions/replace.html.md

```diff
@@ -20,9 +20,18 @@ each occurrence with a given replacement string.
 replace(string, substring, replacement)
 ```
 
+If `substring` is wrapped in forward slashes, it is treated as a regular
+expression; the syntax conforms to the [re2 regular expression
+syntax](https://github.com/google/re2/wiki/Syntax) syntax. If using a regular
+expression, `replace` can reference subcaptures in the regular expression by
```

Comment:
It took a few reads for me to fully grok this sentence; I think it could help to be clearer about where each of these things is allowed, perhaps like this:

> If using a regular expression for the `substring` argument, the `replacement` string can incorporate captured strings from the input by using an `$n` sequence, where `n` is the index or name of a capture group.

Otherwise, I think it could be understood as saying that the regex pattern itself can contain backreferences, which re2 specifically doesn't support.

Reply:
This was taken directly from the 0.11 docs, but I do like your version! ✅